### PR TITLE
Expose Hosting Trial: Transition site to Creator Trial on plugins upload page

### PR DESCRIPTION
**Pull request by paulopmt1**
Related to https://github.com/Automattic/dotcom-forge/issues/5097

## Proposed Changes

This PR adds logic to transition a site to the Creator Trial while remaining on the plugins upload page.

* Show `Start your free trial ` CTA on free sites
* Show the trial acknowledge modal
* Add an onTick call back to the `HostingActivateStatus` component that is executed each time the status is fetched
* Transition the site to atomic when the `Start trial` CTA is clicked in the modal

## Testing Instructions

* go to `/plugins/upload/site.wordpress.com` 
* Verify the CTA has "Start your free trial" if the user is eligible 
* Verify the CTA has "Upgrade and continue" if user is not eligible for trial
* Click CTA and start the trial from modal
* Verify site is transition to Creator Trial without leaving the page and you can upload a plugin after

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by vykes-mac** [comment]
We have to test and deploy both in tandem ? This one has a dependence on the other ?

**Comment by paulopmt1** [comment]
> tandem

This is still on POC. I'll make sure to update you as soon as I advance more on it. Currently, it depends on the other. I'm considering splitting even further to have small deliverables and avoid that dependency as much as possible.

**Review by zaguiini — COMMENTED** [review]
I think we're close. I spotted a few issues, but overall it's working.

I don't understand what's the point with the "Your free trial has been requested" notice. If the user clicked and it started transferring to Atomic, I think it's pretty clear that the trial has begun.

It's also saying "Installation in progress," but that might come off as confusing since the user hasn't yet uploaded the zip file for the plugin 😬 Maybe we should hide this copy? It's redundant anyways and the screen feels busy.

image

After the transfer succeeded I saw the wpcomstaging URL on the left but the address bar still shows the wordpress.com one:

image

Clicking the "Plugins" back button after the transfer (that is, if I change my mind about installing a plugin), I see the upsell nudge. We should hide it, right? Refreshing the page makes it go away.

image

The experience feels a bit rough after clicking the start trial button in the modal. There is no loading feedback. We could make the modal persist with the CTA in busy state until we kick off the transfer, for example.

**Comment by vykes-mac** [comment]
> It's also saying "Installation in progress," but that might come off as confusing since the user hasn't yet uploaded the zip file for the plugin 😬 Maybe we should hide this copy? It's redundant anyways and the screen feels busy.

This copy is shown when the site is in transfer exist state, I understood to represents both a plugin being installed and a atomic transfer (installation is in progress) . We could hide the copy but the UI would look a little weird 

image

> Clicking the "Plugins" back button after the transfer (that is, if I change my mind about installing a plugin), I see the upsell nudge. We should hide it, right? Refreshing the page makes it go away.

Yeah it shouldn't be displayed

> The experience feels a bit rough after clicking the start trial button in the modal. There is no loading feedback. We could make the modal persist with the CTA in busy state until we kick off the transfer, for example.

This makes sense

**Comment by vykes-mac** [comment]
> After the transfer succeeded I saw the wpcomstaging URL on the left but the address bar still shows the wordpress.com one:

Have you encountered any prior art about this? I thinking of updating the url after requesting the updated sites details with the new URL but it feels hacky.

**Comment by zaguiini** [comment]
>Have you encountered any prior art about this?

Not really. I thought the hosting config transition did it automatically, but apparently it doesn't. Would be consistent to have it, though.

>I thinking of updating the url after requesting the updated sites details with the new URL but it feels hacky.

I guess we can introduce a mechanism that verifies if the site URL changed and add a replace entry in history to update it, globally, Calypso-wise. I think this is part of a separate PR, though.

**Comment by zaguiini** [comment]
>We could hide the copy but the UI would look a little weird

Can we hide the whole card? They're aware that the transition is in progress.

File: client/my-sites/hosting/test/hosting-activate-status.jsx [hunk 1]
```diff
@@ -62,20 +62,20 @@ describe( 'index', () => {
 
 		render(
 			<Provider store={ store }>
-				<HostingActivateStatus context="plugins" />
+				<HostingActivateStatus context="plugin" />
```
Comment (zaguiini) [inline client/my-sites/hosting/test/hosting-activate-status.jsx:65]: Any reason to change it from plural to singular?

**Review by zaguiini — COMMENTED** [review]
Should we remove this banner if the user is eligible for the free trial?

image

After starting the trial I did not see the plugin upload drop area:

https://github.com/Automattic/wp-calypso/assets/26530524/0c65ee2e-a9b6-4fbc-84ca-ebcb07f7db6a

There's also some delay after starting the trial. I see the "installation in progress" notice in the background before the modal disappears. No biggie, but it'd be a nice touch to keep them in sync

It's also annoying that we're showing two nudges in the sidebar 😬

File: client/my-sites/hosting/test/hosting-activate-status.jsx (hunk 1 above)
Comment (vykes-mac) [reply client/my-sites/hosting/test/hosting-activate-status.jsx:65]: the `HostingActivateStatus` component `switch` uses singular, it was causing the test to fail as it always goes to the default case.

Comment (vykes-mac) [reply client/my-sites/hosting/test/hosting-activate-status.jsx:65]: > After starting the trial I did not see the plugin upload drop area:

This had always worked for me and show the dropzone, I'll figure why it didn't work

**Comment by vykes-mac** [comment]
> Should we remove this banner if the user is eligible for the free trial?

I've hidden that card for free trails

> After starting the trial I did not see the plugin upload drop area:

This should be now rectified

> There's also some delay after starting the trial. I see the "installation in progress" notice in the background before the modal disappears. No biggie, but it'd be a nice touch to keep them in sync

I've made some changes to keep it in sybc

> It's also annoying that we're showing two nudges in the sidebar 😬

I refetched the JITM so the old domain related message should be cleared after the site goes atomic

**Comment by zaguiini** [comment]
This PR is this close 🤏 to get merged.

I see the transition notice forever :( When I update the page, it goes away and I see the drop zone:

image

Wondering about this blank space here. Maybe we can keep the activation notice that transfers the domain? Display something else? Like, stats about the most used plugins, etc? Something that will entice them to use plugins.

image

**Comment by vykes-mac** [comment]
> Display something else? 

We could show the dropzone grayed out similar to the themes page? 
image

> I see the transition notice forever :( When I update the page, it goes away and I see the drop zone:

hmm, did you try it on an older site? What happen when you test it on a newly created site?

**Comment by zaguiini** [comment]
>We could show the dropzone grayed out similar to the themes page?

That's interesting 👍 

>hmm, did you try it on an older site? What happen when you test it on a newly created site?

This was a fresh site created in `/start`.

**Comment by zaguiini** [comment]
@vykes-mac I'm able to drag and drop a plugin to the area while the transfer is happening 😬 

This time the transfer worked; however I saw a blank card between the transfer notice removal and the drop area being displayed.

**Review by zaguiini — APPROVED** [review]
Worked fine this time!


https://github.com/Automattic/wp-calypso/assets/26530524/0bdc0145-eab5-405c-abee-609d024a4545

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/11029188

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @paulopmt1, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `Error starting your free trial.`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/11029188).

**Pull request merged**